ql/src/java/org/apache/hadoop/hive/ql/ddl/table/partition/alter/AlterTableAlterPartitionOperation.java

```diff
@@ -92,22 +91,23 @@ private void checkPartitionValues(Table tbl, int colIndex) throws HiveException
     Converter converter = ObjectInspectorConverters.getConverter(
         PrimitiveObjectInspectorFactory.javaStringObjectInspector, outputOI);
 
-    Set<Partition> partitions = context.getDb().getAllPartitionsOf(tbl);
-    for (Partition part : partitions) {
-      if (part.getName().equals(context.getConf().getVar(HiveConf.ConfVars.DEFAULTPARTITIONNAME))) {
-        continue;
-      }
-
+    List<String> partNames = context.getDb().getPartitionNames(tbl.getDbName(),
```

Comment:
Good work on eliminating List<Partitions> call because it is absolutely not required.
One suggestion I have is, can we use listParititonNamesRequest() or listPartitionNames(catname, dbName, tableName, maxParts)?

Reply:
Also, will the listPartitions() call ignore 'HiveConf.ConfVars.DEFAULTPARTITIONNAME' and return only the true partition names?

Reply:
The `Hive` class(context.getDb()) doesn't provide direct access to `listParititonNamesRequest` or `listPartitionNames(catname...)`,  the `listParititonNames` in Hive will catch `NoSuchObjectException` and return an empty list in case no partition exists, so I prefer Hive#listPartitionNames here.

By default, all listPartitions APIs would not ignore the default partition, except when provide a valid filter.